src/xs2arr/model.py

```diff
@@ -138,6 +123,29 @@ def fit(
 
         return results
 
+    @staticmethod
+    def _validate_and_prepare_T_grid(
+        T_grid: np.ndarray | None, mean_E_grid: np.ndarray | None
+    ):
+        if T_grid is not None and mean_E_grid is not None:
+            raise ValueError("Only one of T_grid or mean_E_grid must be provided")
+        if T_grid is None and mean_E_grid is None:
+            T_grid = np.linspace(start=0.001, stop=6.0, num=1000, dtype=float)
+        if mean_E_grid is not None:
+            T_grid = 2.0 * mean_E_grid / 3.0
+        if isinstance(T_grid, list | tuple):
+            T_grid = np.array(T_grid, dtype=float)
+        if not isinstance(T_grid, np.ndarray):
+            raise TypeError("T_grid must be of type ndarray")
```

Comment:
Rather than checking the input type, which might miss other array-like types, you can use `np.asarray` to cast into a NumPy array, and if it's already a NumPy array it'll do nothing:

```suggestion
        T_grid = np.asarray(T_grid, dtype=float)
```

Reply:
This is great and I never thought of it (despite using asarray elsewhere in the code). I've always wanted a way to allow a user to specify something array-like but don't want to do all the manual checking myself - thanks numpy.
Thanks for suggestion - will implement